qiskit/opflow/gradients/circuit_gradients/circuit_gradient.py

```diff
@@ -66,3 +68,28 @@ def convert(
             ValueError: If ``params`` contains a parameter not present in ``operator``.
         """
         raise NotImplementedError
+
+    @staticmethod
+    def _unroll_to_supported_operations(
+        circuit: QuantumCircuit, supported_gates: Set[str]
+    ) -> QuantumCircuit:
+        """Unroll the given circuit into a gate set for which the gradients may be computed.
+
+        Args:
+            circuit: Quantum circuit to be unrolled into supported operations.
+
+        Returns:
+            Quantum circuit which is unrolled into supported operations.
+
+        """
+        unique_ops = set(circuit.count_ops().keys())
+        if not unique_ops.issubset(supported_gates):
+            try:
+                unroller = Unroller(list(supported_gates))
```

Comment:
I don't think you want to use the unroller here, it won't really give you the flexibility to convert between basis sets as it works on the fixed definitions of each gate. You should probably use the `BasisTranslator` here (or maybe synthesis like https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/transpiler/preset_passmanagers/level1.py#L169-L176 although that will be slower and not cover all the available basis)

Reply:
Thanks for the suggestion. I will look into this with @Cryoris.

Reply:
This is mainly to fix cases where gates were wrapped into a gate/instruction and must be unrolled again, but if we used the basis translator we could also cover more cases 👍🏻 Do you know if the BasisTranslator is as fast as the Unroller if it just needs to decompose and not translate?

Reply:
After a discussion with @Cryoris, I changed an unroller to a transpiler.